Index: fat.c
===================================================================
--- fat.c.orig
+++ fat.c
@@ -722,7 +722,7 @@ static int old_fat_read(Fs_t *This, unio
 	if(check_media_type(This,boot, tot_sectors))
 		return -1;
 
-	if(This->num_clus >= FAT12) {
+	if(This->num_clus >= FAT12 || config_fat_bits == 16) {
 		set_fat16(This);
 		/* third FAT byte must be 0xff */
 		if(!mtools_skip_check && readByte(This, 3) != 0xff)
